src/otx/algorithms/visual_prompting/tasks/inference.py

```diff
@@ -86,9 +90,13 @@ def __init__(self, task_environment: TaskEnvironment, output_path: Optional[str]
         # Hyperparameters.
         self._work_dir_is_temp = False
         self.output_path = output_path
+        self.mode = "train"
+        if task_environment.model is not None and task_environment.model.train_dataset is None:
+            self.mode = "export"
         if self.output_path is None:
             self.output_path = tempfile.mkdtemp(prefix="otx-visual_prompting")
             self._work_dir_is_temp = True
+            self.mode = "inference"
```

Comment:
I think this kind of branch looks risky. Can this task get mode from cli? and why inference task care train mode?

Reply:
You're right. I think we can use `mode` from `train.py` like below.
```python
task = task_class(task_environment=environment, output_path=str(config_manager.output_path / "logs"), mode=config_manager.mode)
```
But this change will affect any other tasks so I'm not sure if it is okay to handle this part.

Reply:
I think directly set mode from cli is more safe, but leave it as it is.